File: mobile/lib/services/notification_preferences_service.dart
import 'api_client.dart';

/// 알림 카테고리별 on/off (mobile = 사용자 측).
///
/// 백엔드 (Phase L):
///   GET  /api/users/me/notification-preferences
///   PUT  /api/users/me/notification-preferences/{category}
class NotificationPreference {
  final String category;
  final String label;
  final bool enabled;

  const NotificationPreference({
    required this.category,
    required this.label,
    required this.enabled,
  });

  factory NotificationPreference.fromJson(Map<String, dynamic> j) =>
      NotificationPreference(
        category: j['category'] as String,
        label:    j['label']    as String,
        enabled:  j['enabled']  as bool,
      );
}

class NotificationPreferencesService {
  NotificationPreferencesService._();
  static final NotificationPreferencesService instance =
      NotificationPreferencesService._();
  factory NotificationPreferencesService() => instance;

  /// 카탈로그 + 현재 enabled 상태.
  Future<List<NotificationPreference>> list() async {
    final res = await ApiClient.instance
        .get('/api/users/me/notification-preferences');
    final list = (res['preferences'] as List? ?? []);
    return list
        .map((e) => NotificationPreference.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  /// 특정 카테고리 on/off.
  Future<void> set(String category, bool enabled) async {
    await ApiClient.instance.put(
      '/api/users/me/notification-preferences/$category',
      body: {'enabled': enabled},
    );
  }
}

File: provider-web/src/services/notification/NotificationPreferencesService.js
/**
 * NotificationPreferencesService — 알림 카테고리별 on/off (Phase L).
 *
 * 백엔드 (sub_type='facility' 토큰):
 *   GET  /api/facility/me/notification-preferences
 *   PUT  /api/facility/me/notification-preferences/{category}
 *
 * 사용자(mobile) 측은 /api/users/me/notification-preferences — 별도.
 */
import apiClient from '../apiClient';

const NotificationPreferencesService = {
  /** 카탈로그 + 현재 enabled 상태. */
  list() {
    return apiClient.get('/api/facility/me/notification-preferences');
  },

  /** 특정 카테고리 on/off. */
  set(category, enabled) {
    return apiClient.put(
      `/api/facility/me/notification-preferences/${encodeURIComponent(category)}`,
      { enabled },
    );
  },
};

export default NotificationPreferencesService;
